app/controllers/admin/merchants_controller.rb

```diff
@@ -14,14 +14,29 @@ def edit
   def update
     merchant = Merchant.find(params[:id])
 
-    if merchant.update({name: params[:name]})
+    if params.key?(:name)
+      if merchant.update({name: params[:name]})
+        merchant.save
+        redirect_to admin_merchant_path(merchant.id)
+        flash[:alert] = "Merchant Successfully Updated"
+      else
+        redirect_to admin_merchant_path(merchant.id)
+        flash[:alert] = "Merchant Unsuccessfully Updated"
+      end
+    end
 
-      merchant.save
-      redirect_to admin_merchant_path(merchant.id)
-      flash[:alert] = "Merchant Successfully Updated"
-    else
-      redirect_to admin_merchant_path(merchant.id)
-      flash[:alert] = "Merchant Unsuccessfully Updated"
+    if params.key?(:enabled) 
```

Comment:
In my opinion, the two separate if statements can be connected to an if/elsif/else. Line 28 would be an elsif params.key?(:enabled) and be indented the same as line 17.

Reply:
agreed, this works